custom_components/wyzeapi/sensor.py

```diff
@@ -499,152 +498,130 @@ async def async_added_to_hass(self) -> None:
 
+    _attr_entity_category = EntityCategory.DIAGNOSTIC
+    _attr_entity_registry_enabled_default = False
+
     @property
     def name(self) -> str:
         """Return the name of the sensor."""
-        return f"{self._device.nickname} IP Address"
+        return f"{self._irrigation.nickname} IP Address"
 
     @property
     def unique_id(self) -> str:
         """Return a unique ID for the sensor."""
-        return f"{self._device.mac}-ip"
+        return f"{self._irrigation.mac}-ip"
 
     @property
     def native_value(self) -> str:
         """Return the IP address."""
-        return self._device.IP
+        return self._irrigation.IP
 
 
-class WyzeIrrigationWiFiMAC(WyzeIrrigationBaseSensor):
-    """Representation of a Wyze Irrigation WiFi MAC sensor."""
+class WyzeIrrigationSSID(WyzeIrrigationBaseSensor):
+    """Representation of a Wyze Irrigation SSID sensor."""
+
+    _attr_entity_category = EntityCategory.DIAGNOSTIC
+    _attr_entity_registry_enabled_default = False
 
     @property
     def name(self) -> str:
         """Return the name of the sensor."""
-        return f"{self._device.nickname} WiFi MAC"
+        return f"{self._irrigation.nickname} SSID"
 
     @property
     def unique_id(self) -> str:
         """Return a unique ID for the sensor."""
-        return f"{self._device.mac}-wifi-mac"
+        return f"{self._irrigation.mac}-ssid"
 
     @property
     def native_value(self) -> str:
-        """Return the WiFi MAC address."""
-        return self._device.wifi_mac
+        """Return the SSID."""
+        return self._irrigation.ssid
 
 
 class WyzeIrrigationSerialNumber(WyzeIrrigationBaseSensor):
     """Representation of a Wyze Irrigation Serial Number sensor."""
 
-    @property
-    def name(self) -> str:
-        """Return the name of the sensor."""
-        return f"{self._device.nickname} Serial Number"
-
-    @property
-    def unique_id(self) -> str:
-        """Return a unique ID for the sensor."""
-        return f"{self._device.mac}-serial-number"
-
-    @property
-    def native_value(self) -> str:
-        """Return the serial number."""
-        return self._device.sn
-
-
-class WyzeIrrigationSSID(WyzeIrrigationBaseSensor):
-    """Representation of a Wyze Irrigation SSID sensor."""
+    _attr_entity_category = EntityCategory.DIAGNOSTIC
+    _attr_entity_registry_enabled_default = False
 
     @property
     def name(self) -> str:
         """Return the name of the sensor."""
-        return f"{self._device.nickname} SSID"
+        return f"{self._irrigation.nickname} Serial Number"
 
     @property
     def unique_id(self) -> str:
         """Return a unique ID for the sensor."""
-        return f"{self._device.mac}-ssid"
+        return f"{self._irrigation.mac}-serial-number"
 
     @property
     def native_value(self) -> str:
-        """Return the SSID."""
-        return self._device.ssid
+        """Return the serial number."""
+        return self._irrigation.sn
```

Comment:
This can be eliminated by just adding `serial_number=self._irrigation.serial` to DeviceInfo in the base class